src/Common/src/Interop/Unix/libcurl/Interop.SafeCurlHandle.cs

```diff
@@ -37,6 +40,19 @@ protected override bool ReleaseHandle()
 
         internal sealed class SafeCurlMultiHandle : SafeHandle
         {
+            private readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
+            private readonly List<int> _fdList = new List<int>();
```

Comment:
@vijaykota For small application, this would be in the 10's.  But for server scalability scenarios (such as we have seen in large deployments of .NET like Office365, etc., there could be 10's of thousands (i.e. 10,000 -> 100,000) requests in flight to various endpoints.

So, general guidance is to think PERFORMANCE when coding this library.